### ISSUE TITLE
Further whitespace issues

### ISSUE THREAD

**Issue by alvinlindstam** [rrrene/html_sanitize_ex#14]
As noted in https://github.com/rrrene/html_sanitize_ex/issues/4, mochiweb_html ignores whitespace between tags. The fix (replacing the input string using a regexp) doesn't really fix the issues though.

In this example, the whitespace is not only spaces, but also a newline. Space is not the only [space character](https://www.w3.org/TR/html5/infrastructure.html#space-character) in HTML.

```
iex(39)> HtmlSanitizeEx.basic_html("<a href=\"almost\">on my mind</a>  <a href=\"almost\">all day long</a>")
"<a href=\"almost\">on my mind</a> <a href=\"almost\">all day long</a>"
iex(40)> HtmlSanitizeEx.basic_html("<a href=\"almost\">on my mind</a>  \n<a href=\"almost\">all day long</a>")
"<a href=\"almost\">on my mind</a><a href=\"almost\">all day long</a>"
```

In this example, mochiweb properly parses the textarea contents, which is later escaped on output. But since we regex-replaced the space with `&#32;`, that sequense is also escaped:

```
iex(50)> HtmlSanitizeEx.html5("<textarea> <script></script></textarea>") 
"<textarea>&amp;#32;&lt;script&gt;&lt;/script&gt;</textarea>"
```

The first issue could probably be solved with an extended regexp to match all space characters, while the second one could only be solved by making the parser keep all text nodes.

**Comment by alvinlindstam** [comment]
It seems that I was a little quick on the keys, https://github.com/rrrene/html_sanitize_ex/issues/12 seems to be about the same issue.

**Comment by rrrene** [comment]
Mmh, that's not good. I did not think of `<textarea>` when contemplating fixes to this problem. `<pre>` might very well have the same problems.

Thanks for reporting! :+1:

**Comment by rrrene** [comment]
@alvinlindstam I know it has been a while, but I just published `v1.3.0-rc1` which should fix this issue. If you are still using `html_sanitize_ex`, please try it out! :+1:
